tools/pipelines/templates/include-publish-npm-package-deployment.yml

```
@@ -51,28 +51,46 @@ jobs:
   variables:
     version: $[ stageDependencies.build.build.outputs['SetVersion.version']]
     isLatest: $[ stageDependencies.build.build.outputs['SetVersion.isLatest']]
+  templateContext:
+    type: releaseJob
+    isProduction: ${{ not(eq(variables['testBuild'], 'true')) }}
+    # This section downloads initial required artifacts/data for this job.
+    inputs:
+      - input: pipelineArtifact
+        pipeline: buildTools-resource
+        artifactName: pack
+        targetPath: $(Pipeline.Workspace)/buildTools-zip
+
+      - input: pipelineArtifact
+        artifactName: pack
+        buildType: current
+        targetPath: $(Pipeline.Workspace)/pack
   strategy:
     runOnce:
         deploy:
           steps:
-          - checkout: self
-            clean: true
-            persistCredentials: true # Necessary for creation of git tags to work
-          - download: current
-            artifact: pack
           - template: /tools/pipelines/templates/include-use-node-version.yml@self
-          - template: /tools/pipelines/templates/include-install-build-tools.yml@self
-            parameters:
-              buildDirectory: ${{ parameters.buildDirectory }}
-              buildToolsVersionToInstall: ${{ parameters.buildToolsVersionToInstall }}
-              pnpmStorePath: ${{ parameters.pnpmStorePath }}
+
+          - task: Bash@3
+            name: InstallBuildToolsFromTarball
+            displayName: Install Fluid Build Tools from artifact tarball
+            inputs:
+              targetType: 'inline'
+              workingDirectory: '$(Pipeline.Workspace)/buildTools-zip/tarballs'
+              script: |
+                echo "Listing files in directory: $(pwd)"
```

Best practice in Bash tasks so the first error causes the task to fail; otherwise something can fail and if a later command in the script succeeds, it will look like the task succeeded.
```suggestion
                set -eu -o pipefail
                echo "Listing files in directory: $(pwd)"
```

---

Updated.